trellowarrior/main.py

```diff
@@ -16,454 +17,708 @@
+    seconds = args.delai_between_sync * 60
+    if seconds > 0:
+        try:
+            while True:
+                logger.info("Sync done. will wait {} minutes before next one "
+                            "(press ctrl-c to stop)".format(args.delai_between_sync))
+                sleep(seconds)
+                tw.invalid_trello_cache()
+                tw.sync(projects)
+        except KeyboardInterrupt:
+            pass
 
 def authenticate(args):
     """
     Function called by the authenticate subcommand to perform the initial authentication
 
     :args: Command line arguments. Must contain the following attributes : 'config', 'api_key', 'api_key_secret', 'name'
     """
-    if os.path.exists(args.config):
-        print("File already exists, aborting")
-
-    oauth_token = create_oauth_token(expiration=args.expiration, key=args.api_key, secret=args.api_key_secret, name=args.name)
-    conf = RawConfigParser()
-    conf.set("DEFAULT", "trello_api_key", args.api_key)
-    conf.set("DEFAULT", "trello_api_secret", args.api_key_secret)
-    conf.set("DEFAULT", "trello_token", oauth_token['oauth_token'])
-    conf.set("DEFAULT", "trello_token_secret", oauth_token['oauth_token_secret'])
-    with open(args.config, 'w') as f:
-        conf.write(f)
+    tw = TrelloWarrior(config_file=args.config)
+    tw.authenticate(args.expiration, args.api_key, args.api_key_secret, args.name)
+
+
+def new_project(args):
+    """
+    Function called by the new subcommand to add a project to the config
+
+    :args: Command line arguments. Must contain the following attributes : 'config', 'name', 'tw_name', 'board_name',
+            'todo', 'doing', 'done', 'no_sync'
+    """
+    tw = TrelloWarrior(config_file=args.config)
+    tw.new_project(args.name, args.tw_name, args.board_name, args.todo, args.doing, args.done, args.no_sync)
+
 
 def main():
     parser = argparse.ArgumentParser(description="Bidirectional sync between trello and taskwarrior")
     parser.add_argument('-c', '--config', metavar='value', help='custom configuration file path')
+    parser.add_argument('-v', '--verbose', help='Increase verbosity. Can be repeated up to 2 times', action='count')
     subparsers = parser.add_subparsers()
     sync_parser = subparsers.add_parser('sync', help="Synchronize trello and taskwarrior")
-    sync_parser.add_argument("projects", nargs='*', help="List of projects to synchronize. If empty will synchronize all projects listed in the configuration file")
+    sync_parser.add_argument("projects", nargs='*',
+                             help="List of projects to synchronize. If empty will synchronize all projects listed in "
+                                  "the configuration file")
+    sync_parser.add_argument("--delai-between-sync", '-d', metavar='N', default=0, type=int,
+                             help="Launch the sync process every N minutes. If set to 0 (default), perform one sync and"
+                                  " exit.")
```

Comment:
I don't see this option. You can configure a timer in cron or in systemd to execute automagically `trellowarrior`. I think that can go without it.

Reply:
I'm thinking of running it in a docker container, without cron nor systemd. And I like the simplicity of running it in a spare terminal on my desktop...

Reply:
In this case I think that is better use `-d`, `--daemon` with a default delay (1 hour for example) and leave the delay setting to config file. :wink:

Reply:
I will do that. I'm also thinking of ditching INI format in favor of yaml. Do you have any problem with that (I will of course provide a tool for conversion) ? I can also add support for both formats, but I'm not sure it's a good idea.

Reply:
What made me think of yaml was the part about the tags/color correspondence. Splitting on ',' and then on '=' is not pretty and configuration is not really readable.

If an external library is a no go, then json (with pretty print) could be an alternative.

Reply:
Cannot be `trello_tag_[name] = [color]`?

Reply:
Yes, that's a real missing feature. But the readability of the code for the parsing and writing would be greatly increased.